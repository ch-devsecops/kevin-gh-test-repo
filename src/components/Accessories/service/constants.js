import PropTypes from 'prop-types';

export const variant1 = 'variant1';
export const variant2 = 'variant2';
export const variant3 = 'variant3';

export const ACCESSORY_TYPE_LABEL_ALL = 'all';
export const ALL_TAB = 'all';
export const INTERIOR_TAB = 'interior';
export const EXTERIOR_TAB = 'exterior';
export const PACKAGES_TAB = 'packages';
export const TRIM_DROPDOWN_KEY = 'trim';
export const TRANSMISSION_DROPDOWN_KEY = 'transmission';
export const ACCESSORY_TYPE_DROPDOWN_KEY = 'accessoryType';
export const ACCESSORIES_WRAPPER_INTERACTION_TYPE = 'accessories - accessory filter';
export const ACCESSORIES_CARD_INTERACTION_TYPE = 'accessory details';

export const gtmType = PropTypes.shape({
  'data-gtm-model': PropTypes.string,
  'data-gtm-trim': PropTypes.string,
  'data-gtm-body-style': PropTypes.string,
});
