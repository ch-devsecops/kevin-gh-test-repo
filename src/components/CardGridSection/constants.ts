const CAROUSEL_SCROLL_WIDTH = 271; // Item Width (259px) + Carousel X Padding (12px)

export default CAROUSEL_SCROLL_WIDTH;
