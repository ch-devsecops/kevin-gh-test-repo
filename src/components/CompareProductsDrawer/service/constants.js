export const variant1 = 'variant1';
export const variant2 = 'variant2';
export const variant3 = 'variant3';

export const ALIGN_CENTER = 'center';
export const ZERO = 'zero';
export const NUMBER_OF_CARDS = 4;
