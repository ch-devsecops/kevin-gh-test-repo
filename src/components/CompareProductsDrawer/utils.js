import { flattenDeep } from 'lodash/array';
import getValueByKey from '../../utils/getValueByKey';

/**
 * returns the miniature product card data for selected products to compare
 * @param detIds
 * @param data
 * @returns {Array}
 */
function getModels(detIds, data) {
  return flattenDeep(
    data?.map(category =>
      category?.subCategories?.map(subCategory =>
        subCategory?.models?.filter(model => detIds.includes(model?.detIdentifier?.value)),
      ),
    ),
  );
}

/**
 * maps marine web catalog to the expected data structure by the UI component
 * @param outboards
 * @returns {*}
 */
export function getMarineMappedCatalogData(outboards) {
  return outboards?.map(item => ({
    subCategories: [
      {
        id: item?.id,
        models: item?.models?.map(model => ({
          ...model,
          thumbnail: {
            ...model?.primaryImage?.item?.value,
            url: model?.primaryImage?.item?.value.src,
            alt: { value: model?.primaryImage?.item?.value.alt },
          },
        })),
      },
    ],
  }));
}

/**
 * maps engine web catalog to the expected data structure by the UI component
 * @param series
 * @returns {Array}
 */
export function getEngineMappedCatalogData(series) {
  return flattenDeep(
    series?.map(s =>
      s.crankshafts?.map(crankshaft => ({
        subCategories: [
          {
            id: crankshaft?.id,
            models: crankshaft?.models?.map(model => ({
              ...model,
              thumbnail: getValueByKey(model, 'images')?.[0],
            })),
          },
        ],
      })),
    ),
  );
}

export default getModels;
