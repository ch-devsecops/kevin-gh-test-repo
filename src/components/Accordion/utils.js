import { ENGINE_SITE_NAME, MARINE_SITE_NAME } from '../../utils/constants';
import { getGtmTagValue } from '../../utils/gtmEvents';

/**
 * Creates GTM attributes for a productName
 * @param {string} productName
 * @param {object} gtmTags
 * @returns gtmAttributes
 */
export const getAccordionGtmTagsByProductName = ({ appName, gtmTags }) => {
  switch (appName) {
    case MARINE_SITE_NAME:
      return Object.freeze({
        'data-gtm-interaction-type': 'primary model nav',
        'data-gtm-component-type': getGtmTagValue(gtmTags?.componentName),
        'data-gtm-title': getGtmTagValue(gtmTags?.gtmTitle),
        'data-gtm-accordion-level': 'L1',
      });
    case ENGINE_SITE_NAME:
      return Object.freeze({
        'data-gtm-interaction-type': getGtmTagValue(gtmTags?.interactionType),
        'data-gtm-title': getGtmTagValue(gtmTags?.gtmTitle),
        'data-gtm-component-type': getGtmTagValue(gtmTags?.componentName),
        'data-gtm-body-style': getGtmTagValue(gtmTags?.bodyStyle),
        'data-gtm-model': getGtmTagValue(gtmTags?.modelName),
        'data-gtm-trim': getGtmTagValue(gtmTags?.trimName),
      });
    default:
      return {};
  }
};

export const getGtmTags = (category, componentName) =>
  Object.freeze({
    'data-gtm-category': getGtmTagValue(category),
    'data-gtm-component-type': getGtmTagValue(componentName),
  });
