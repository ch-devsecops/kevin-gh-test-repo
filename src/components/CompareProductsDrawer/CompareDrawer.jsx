import React, { useContext } from 'react';
import { Box, Button } from '@honda-canada/design-system-react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import themeStyles from './CompareDrawer.styles';
import DrawerCards from './DrawerCards';
import LocalStorageContext from '../LocalStorageContext/LocalStorageContext';
import { getComparePageUrl } from './service/utils';
import ScrollableDrawerCards from './ScrollableDrawerCards';
import Context from './service/Context';
import { gtmCompareDetails } from '../../utils/gtmEvents';

const ButtonGroup = themeStyles.apply(Box, 'ButtonGroup');
const SecondaryButton = themeStyles.apply(Button, 'Button');
const PrimaryButton = themeStyles.apply(Button, 'Button');
const DrawerWrapper = themeStyles.apply(Box, 'DrawerWrapper');

const CompareDrawer = ({ compareProducts, isDrawerOpen, setShowModal, gtmComponentName }) => {
  const { deleteToCompareProducts } = useContext(LocalStorageContext);
  const { t } = useTranslation();

  const { styles, hasDifferentProductLines } = useContext(Context) || {};
  const { isScrollable, buttonFontSize } = styles || {};

  const compareButtonHandler = event => {
    gtmCompareDetails(gtmComponentName);
    if (!hasDifferentProductLines) return;

    if (!getComparePageUrl(compareProducts) && typeof setShowModal === 'function') {
      event.preventDefault();
      setShowModal(true);
    }
  };

  const compareCount = compareProducts?.length;

  const compareBtnLabel = t('Shared.Common.drawerCompareButton', { count: compareCount || 0 });

  return (
    <DrawerWrapper data-testid="compare-drawer-products" isVisible={isDrawerOpen}>
      {isScrollable ? (
        <ScrollableDrawerCards compareProducts={compareProducts} />
      ) : (
        <DrawerCards compareProducts={compareProducts} />
      )}
      <ButtonGroup>
        <PrimaryButton
          as="a"
          href={hasDifferentProductLines ? t(getComparePageUrl(compareProducts)) : t('Shared.Common.comparePageUrl')}
          styling="primary"
          data-testid="compare-button"
          data-testvalue={compareCount}
          onClick={event => compareButtonHandler(event)}
          fontSize={buttonFontSize}
        >
          {compareBtnLabel}
        </PrimaryButton>
        <SecondaryButton
          data-testid="clear-button"
          styling="secondary"
          onClick={deleteToCompareProducts}
          fontSize={buttonFontSize}
        >
          {t('Shared.Common.clearAllButton')}
        </SecondaryButton>
      </ButtonGroup>
    </DrawerWrapper>
  );
};

CompareDrawer.propTypes = {
  compareProducts: PropTypes.arrayOf(PropTypes.shape({})),
  isDrawerOpen: PropTypes.bool,
  setShowModal: PropTypes.func,
  gtmComponentName: PropTypes.string,
};

export default CompareDrawer;
