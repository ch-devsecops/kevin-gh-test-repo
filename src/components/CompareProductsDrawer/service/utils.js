import { flattenDeep } from 'lodash/array';
import getValueByKey from '../../../utils/getValueByKey';
import { MC_PRODUCT_NAME } from '../../../utils/constants';

/**
 * returns the miniature product card data for selected products to compare
 * @param detIds - model DET IDs
 * @param data - product catalog data
 * @returns {Array}
 */
export function getEngineModels(detIds, data) {
  return flattenDeep(
    data?.map(category =>
      category?.subCategories?.map(subCategory =>
        subCategory?.models?.filter(model => detIds.includes(model?.detIdentifier?.value)),
      ),
    ),
  );
}

/**
 * returns the miniature product card data for selected products to compare
 * @param detIds - transmission DET IDs
 * @param data - mapped product catalog data, from getPspMappedCatalogData
 * @returns {Array}
 */
export function getPspModels(detIds, data) {
  const models = flattenDeep(
    data?.map(category =>
      category?.subCategories?.map(subCategory =>
        subCategory?.models?.filter(model => model.transmissions?.some(item => detIds?.includes(item))),
      ),
    ),
  );
  // ensures models are returned in the same order as the provided detIds
  return detIds.map(detId => models.find(model => model.transmissions.includes(detId))).filter(id => !!id);
}

/**
 * maps marine web catalog to the expected data structure by the UI component
 * @param outboards
 * @returns {*}
 */
export function getMarineMappedCatalogData(outboards) {
  return outboards?.map(item => ({
    subCategories: [
      {
        id: item?.id,
        models: item?.models?.map(model => ({
          ...model,
          thumbnail: {
            ...model?.primaryImage?.item?.value,
            url: model?.primaryImage?.item?.value.src,
            alt: { value: model?.primaryImage?.item?.value.alt },
          },
        })),
      },
    ],
  }));
}

/**
 * maps engine web catalog to the expected data structure by the UI component
 * @param series
 * @returns {Array}
 */
export function getEngineMappedCatalogData(series) {
  return flattenDeep(
    series?.map(s =>
      s.crankshafts?.map(crankshaft => ({
        subCategories: [
          {
            id: crankshaft?.id,
            models: crankshaft?.models?.map(model => ({
              ...model,
              thumbnail: getValueByKey(model, 'images')?.[0],
            })),
          },
        ],
      })),
    ),
  );
}

/**
 * Returns a new array of vehicles with their associated product line.
 *
 * @function
 * @param {Object} vehicleData - The source data containing product lines and their associated vehicle types.
 * @param {string} vehicleData[].name - The name of the product line.
 * @param {Object[]} vehicleData[].vehicleTypes - The types of vehicles under the product line.
 *
 * @returns {Object[]} - An array of vehicles with added 'productLine' property.
 */
const getVehiclesWithProductLine = vehicleData => {
  if (!vehicleData || typeof vehicleData !== 'object') return [];

  let result = [];
  Object.values(vehicleData).forEach(({ name: productLine, vehicleTypes }) => {
    if (Array.isArray(vehicleTypes)) {
      const vehicles = vehicleTypes.map(vehicleType => ({ ...vehicleType, productLine }));
      result = result.concat(vehicles);
    }
  });

  return result;
};
/**
 * maps psp web catalog to the expected data structure by the UI component
 * @param categories
 * @returns {Array}
 */
export const getPspMappedCatalogData = fieldsData => {
  const vehicles = getVehiclesWithProductLine(fieldsData);
  return flattenDeep(
    vehicles?.map(vehicle => {
      const productline = vehicle?.productLine === MC_PRODUCT_NAME ? 'Mc' : 'AtvSxs';
      return vehicle?.categories?.map(category =>
        category.modelYears?.map(modelYear =>
          modelYear?.models?.map(model => ({
            subCategories: [
              {
                id: model?.sotId,
                models: model?.trims
                  ? model.trims.map(trim => ({
                      ...trim,
                      modelName: {
                        value: trim?.trimName?.value,
                      },
                      thumbnail: {
                        ...trim?.primaryThumbnail?.item?.value?.images?.[0],
                      },
                      productline,
                      // store all transmission IDs under this trim/model
                      transmissions: trim?.transmissions?.[0]?.items?.map(tran => tran?.detIdentifier?.value),
                    }))
                  : [],
              },
            ],
          })),
        ),
      );
    }),
  );
};

/**
 * Generates a URL for the compare page based on the provided data.
 *
 * @param {Array} data - An array of objects containing product data.
 * @returns {string} The URL for the compare page, or an empty string if the data is invalid.
 */
export const getComparePageUrl = data => {
  if (data.length === 0) {
    return '';
  }
  const firstProductline = data[0].productline;
  const isProductlineSame = data.every(item => item.productline === firstProductline);

  if (!isProductlineSame) return '';

  return `Shared.Common.comparePage${firstProductline}Url`;
};
