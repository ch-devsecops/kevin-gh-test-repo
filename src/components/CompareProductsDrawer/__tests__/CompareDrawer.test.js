import { fireEvent } from '@testing-library/react';
import React from 'react';
import { axe, cleanup, jssRender, screen } from 'test-utils';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import CompareDrawer from '..';
import LocalStorageContext from '../../LocalStorageContext/LocalStorageContext';
import { ENGINE_SITE_NAME } from '../../../utils/constants';

afterEach(cleanup);

const mockData = require('../../../../sample-mock-data/routes/compare-drawer/en.json');
const emptyMockData = require('../../../../sample-mock-data/routes/compare-drawer/empty_en.json');

// todo: improve tests with localstorage mocking
// ref: https://robertmarshall.dev/blog/how-to-mock-local-storage-in-jest-tests/
const localStorageStateMock = {
  toCompareProducts: ['12095', '12096', '12097', '12098'],
  isCompareDrawerOpen: true,
  setToCompareProducts: jest.fn(),
  toggleCompareDrawer: jest.fn(),
  removeFromCompareProducts: jest.fn(),
  deleteToCompareProducts: jest.fn(),
};

const contextMock = {
  site: {
    name: 'Engine',
  },
  pageState: 'normal',
  hondaRestApiHost: 'uat-api.honda.ca',
};

describe('<CompareProductsDrawer>', () => {
  const renderMockCompare = () =>
    jssRender(
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ENGINE_SITE_NAME },
          ...contextMock,
        }}
        componentFactory={{}}
      >
        <LocalStorageContext.Provider value={localStorageStateMock}>
          <CompareDrawer {...mockData} />
        </LocalStorageContext.Provider>
      </SitecoreContext>,
    );

  it('should render correctly', () => {
    renderMockCompare();

    expect(screen.queryByText('GX120')).toBeInTheDocument();
  });

  test('should not render when there is no item in the localstorage', () => {
    jssRender(
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ENGINE_SITE_NAME },
          ...contextMock,
        }}
        componentFactory={{}}
      >
        <LocalStorageContext.Provider value={{ ...localStorageStateMock, toCompareProducts: [] }}>
          <CompareDrawer {...mockData} />
        </LocalStorageContext.Provider>
      </SitecoreContext>,
    );

    expect(screen.queryByText('GX120')).not.toBeInTheDocument();
  });

  test('should return null if there is no fields', () => {
    const { container } = jssRender(
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ENGINE_SITE_NAME },
          ...contextMock,
        }}
        componentFactory={{}}
      >
        <CompareDrawer fields={emptyMockData?.fields} />
      </SitecoreContext>,
    );

    expect(container).toMatchSnapshot();
  });

  test('should empty local storage when cleared', () => {
    const storageMock = { ...localStorageStateMock };
    const { getByTestId, container } = renderMockCompare();
    const clearBtn = getByTestId('clear-button');
    fireEvent.click(clearBtn);
    storageMock.toCompareProducts = [];
    expect(container).toMatchSnapshot();
  });

  test('should have no accessibility violations', async () => {
    const { container } = renderMockCompare();

    expect(container).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });
});
