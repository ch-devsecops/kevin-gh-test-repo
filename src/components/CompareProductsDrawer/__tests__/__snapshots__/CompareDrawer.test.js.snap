// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompareProductsDrawer> should empty local storage when cleared 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 ipaDbT"
    data-testid="compare-product-drawer-container"
  >
    <button
      aria-label="Shared.Common.compareDrawerHeading"
      aria-labelledby="Shared.Common.compareDrawerHeading"
      class="Link-sc-16wd4a0-0 sc-1k44dfv-0 kfScjo"
      data-testid="compare-drawer-toggle"
      tabindex="0"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 jKzALd"
        tabindex="-1"
      >
        <p
          class="sc-1k44dfv-0 fDGMla"
          color="typographyDefault"
          font-family="bold"
          font-weight="regular"
        >
          Shared.Common.compareDrawerHeading
        </p>
      </div>
    </button>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 cZYHRg"
      data-testid="compare-drawer-products"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 kWIvqL"
      >
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 hBrrgo"
            data-testid="compare-drawer-product-card"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 WvtfK"
            >
              <img
                alt="Front of Honda GX120 engine"
                class="sc-1k44dfv-0 sc-1k44dfv-0 jNcVLa"
                draggable="true"
                loading="lazy"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Brands/Engine/21-GX-Series-Listing/GX-120/GX120.jpg?rev=af1f2b74677848d9a78390e8484f38e6"
              />
              <a
                class="sc-1k44dfv-0 sc-1k44dfv-0 jOGOGH"
                color="typographyDefault"
                font-family="default"
                font-weight="regular"
              >
                GX120
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 gyytvc"
              data-testid="compare-drawer-product-card-close-btn"
              tabindex="0"
            />
          </div>
        </div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 hBrrgo"
            data-testid="compare-drawer-product-card"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 WvtfK"
            >
              <img
                alt="Front of Honda GX160 engine"
                class="sc-1k44dfv-0 sc-1k44dfv-0 jNcVLa"
                draggable="true"
                loading="lazy"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Brands/Engine/21-GX-Series-Listing/GX-160/GX160.jpg?rev=7d0192b9fffb4ebf95c8807d00a9c2d6"
              />
              <a
                class="sc-1k44dfv-0 sc-1k44dfv-0 jOGOGH"
                color="typographyDefault"
                font-family="default"
                font-weight="regular"
              >
                GX160
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 gyytvc"
              data-testid="compare-drawer-product-card-close-btn"
              tabindex="0"
            />
          </div>
        </div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 hBrrgo"
            data-testid="compare-drawer-product-card"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 WvtfK"
            >
              <img
                alt="Front of Honda GX200 engine"
                class="sc-1k44dfv-0 sc-1k44dfv-0 jNcVLa"
                draggable="true"
                loading="lazy"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Brands/Engine/21-GX-Series-Listing/GX-200/GX200.jpg?rev=f055a2b356bd4147b24faa9d020d6095"
              />
              <a
                class="sc-1k44dfv-0 sc-1k44dfv-0 jOGOGH"
                color="typographyDefault"
                font-family="default"
                font-weight="regular"
              >
                GX200
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 gyytvc"
              data-testid="compare-drawer-product-card-close-btn"
              tabindex="0"
            />
          </div>
        </div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 hBrrgo"
            data-testid="compare-drawer-product-card"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 WvtfK"
            >
              <img
                alt="Front of Honda GX270 engine"
                class="sc-1k44dfv-0 sc-1k44dfv-0 jNcVLa"
                draggable="true"
                loading="lazy"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Brands/Engine/21-GX-Series-Listing/GX-270/GX270.jpg?rev=e4c6b54fe7544a62868ef340c27459ac"
              />
              <a
                class="sc-1k44dfv-0 sc-1k44dfv-0 jOGOGH"
                color="typographyDefault"
                font-family="default"
                font-weight="regular"
              >
                GX270
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 gyytvc"
              data-testid="compare-drawer-product-card-close-btn"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 gAxfHF"
      >
        <a
          class="Button-pulw6t-0 sc-1k44dfv-0 eSVGfr"
          data-testid="compare-button"
          data-testvalue="4"
          font-size="14px,13px"
          href="Shared.Common.comparePageUrl"
        >
          Shared.Common.drawerCompareButton
        </a>
        <button
          class="Button-pulw6t-0 sc-1k44dfv-0 iTpcTy"
          data-testid="clear-button"
          font-size="14px,13px"
        >
          Shared.Common.clearAllButton
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CompareProductsDrawer> should have no accessibility violations 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 ipaDbT"
    data-testid="compare-product-drawer-container"
  >
    <button
      aria-label="Shared.Common.compareDrawerHeading"
      aria-labelledby="Shared.Common.compareDrawerHeading"
      class="Link-sc-16wd4a0-0 sc-1k44dfv-0 kfScjo"
      data-testid="compare-drawer-toggle"
      tabindex="0"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 jKzALd"
        tabindex="-1"
      >
        <p
          class="sc-1k44dfv-0 fDGMla"
          color="typographyDefault"
          font-family="bold"
          font-weight="regular"
        >
          Shared.Common.compareDrawerHeading
        </p>
      </div>
    </button>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 cZYHRg"
      data-testid="compare-drawer-products"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 kWIvqL"
      >
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 hBrrgo"
            data-testid="compare-drawer-product-card"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 WvtfK"
            >
              <img
                alt="Front of Honda GX120 engine"
                class="sc-1k44dfv-0 sc-1k44dfv-0 jNcVLa"
                draggable="true"
                loading="lazy"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Brands/Engine/21-GX-Series-Listing/GX-120/GX120.jpg?rev=af1f2b74677848d9a78390e8484f38e6"
              />
              <a
                class="sc-1k44dfv-0 sc-1k44dfv-0 jOGOGH"
                color="typographyDefault"
                font-family="default"
                font-weight="regular"
              >
                GX120
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 gyytvc"
              data-testid="compare-drawer-product-card-close-btn"
              tabindex="0"
            />
          </div>
        </div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 hBrrgo"
            data-testid="compare-drawer-product-card"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 WvtfK"
            >
              <img
                alt="Front of Honda GX160 engine"
                class="sc-1k44dfv-0 sc-1k44dfv-0 jNcVLa"
                draggable="true"
                loading="lazy"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Brands/Engine/21-GX-Series-Listing/GX-160/GX160.jpg?rev=7d0192b9fffb4ebf95c8807d00a9c2d6"
              />
              <a
                class="sc-1k44dfv-0 sc-1k44dfv-0 jOGOGH"
                color="typographyDefault"
                font-family="default"
                font-weight="regular"
              >
                GX160
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 gyytvc"
              data-testid="compare-drawer-product-card-close-btn"
              tabindex="0"
            />
          </div>
        </div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 hBrrgo"
            data-testid="compare-drawer-product-card"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 WvtfK"
            >
              <img
                alt="Front of Honda GX200 engine"
                class="sc-1k44dfv-0 sc-1k44dfv-0 jNcVLa"
                draggable="true"
                loading="lazy"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Brands/Engine/21-GX-Series-Listing/GX-200/GX200.jpg?rev=f055a2b356bd4147b24faa9d020d6095"
              />
              <a
                class="sc-1k44dfv-0 sc-1k44dfv-0 jOGOGH"
                color="typographyDefault"
                font-family="default"
                font-weight="regular"
              >
                GX200
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 gyytvc"
              data-testid="compare-drawer-product-card-close-btn"
              tabindex="0"
            />
          </div>
        </div>
        <div
          class="Box-sc-1bsf784-0 iPRsIc"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 hBrrgo"
            data-testid="compare-drawer-product-card"
            tabindex="0"
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 WvtfK"
            >
              <img
                alt="Front of Honda GX270 engine"
                class="sc-1k44dfv-0 sc-1k44dfv-0 jNcVLa"
                draggable="true"
                loading="lazy"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Brands/Engine/21-GX-Series-Listing/GX-270/GX270.jpg?rev=e4c6b54fe7544a62868ef340c27459ac"
              />
              <a
                class="sc-1k44dfv-0 sc-1k44dfv-0 jOGOGH"
                color="typographyDefault"
                font-family="default"
                font-weight="regular"
              >
                GX270
              </a>
            </div>
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 gyytvc"
              data-testid="compare-drawer-product-card-close-btn"
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 gAxfHF"
      >
        <a
          class="Button-pulw6t-0 sc-1k44dfv-0 eSVGfr"
          data-testid="compare-button"
          data-testvalue="4"
          font-size="14px,13px"
          href="Shared.Common.comparePageUrl"
        >
          Shared.Common.drawerCompareButton
        </a>
        <button
          class="Button-pulw6t-0 sc-1k44dfv-0 iTpcTy"
          data-testid="clear-button"
          font-size="14px,13px"
        >
          Shared.Common.clearAllButton
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CompareProductsDrawer> should return null if there is no fields 1`] = `<div />`;
